perun-core/src/test/java/cz/metacentrum/perun/core/entry/GroupsManagerEntryIntegrationTest.java

```diff
@@ -1424,6 +1425,47 @@ public void convertGroupsToRichGroupsWithAttributesWithListOfNamesTest() throws
 		assertEquals("Both lists should be same", Arrays.asList(richGroup1, richGroup2), groupsManagerBl.convertGroupsToRichGroupsWithAttributes(sess, Arrays.asList(group, group2), attrNames));
 	}
 
+	@Test(expected = ExternalyManagedException.class)
+	public void addAndRemoveMemberInSynchronizedGroup() throws Exception {
+		System.out.println(CLASS_NAME + "addAndRemoveMemberInSynchronizedGroup");
+
+		vo = setUpVo();
+		setUpGroup(vo);
+
+		Attribute synchroAttr = new Attribute(perun.getAttributesManager().getAttributeDefinition(sess, "urn:perun:group:attribute-def:def:synchronizationEnabled"));
+		synchroAttr.setValue("true");
+		perun.getAttributesManager().setAttribute(sess, group, synchroAttr);
+
+		Member member = setUpMember(vo);
+		groupsManager.addMember(sess, group, member);
```

Comment:
Hi, test will throw ExternalyManagedException on this line (as expected) and will end. Any further code is not executed. In order to check other use-cases of add/remove methods you must create separate test for each case.